src/lightning_app/components/auto_scaler.py

```diff
@@ -0,0 +1,569 @@
+                    response.raise_for_status()
+                    response = await response.json()
+                    outputs = response["outputs"]
+                    assert len(batch) == len(outputs), f"result has {len(outputs)} items but batch is {len(batch)}"
+                    result = {request[0]: r for request, r in zip(batch, outputs)}
+                    self._responses.update(result)
+        except Exception as ex:
+            result = {request[0]: ex for request in batch}
+            self._responses.update(result)
+
+    async def consumer(self):
+        while True:
+            await asyncio.sleep(0.05)
+
+            has_sent = False
+
+            batch = self._batch[: self.max_batch_size]
+            while batch and (
+                (len(batch) >= self.max_batch_size) or ((time.time() - self._last_batch_sent) > self.timeout_batching)
+            ):
+                has_sent = True
+
+                asyncio.create_task(self.send_batch(batch))
+
+                self._batch = self._batch[self.max_batch_size :]
+                batch = self._batch[: self.max_batch_size]
+
+            if has_sent:
+                self._last_batch_sent = time.time()
+
+    async def process_request(self, data: BaseModel):
+        if not self.servers:
+            raise HTTPException(500, "None of the workers are healthy!")
+
+        request_id = uuid.uuid4().hex
+        request: Tuple = (request_id, data)
+        self._batch.append(request)
+
+        while True:
+            await asyncio.sleep(0.05)
+
+            if request_id in self._responses:
+                result = self._responses[request_id]
+                del self._responses[request_id]
+                _raise_granular_exception(result)
+                return result
+
+    def run(self):
+        if self._server_ready:
+            return
+
+        logger.info(f"servers: {self.servers}")
+
+        self._iter = cycle(self.servers)
+        self._last_batch_sent = time.time()
+
+        fastapi_app = _create_fastapi("Load Balancer")
+        security = HTTPBasic()
+        fastapi_app.global_request_count = 0
+        fastapi_app.num_current_requests = 0
+        fastapi_app.last_processing_time = 0
+        fastapi_app.SEND_TASK = None
+
+        @fastapi_app.on_event("startup")
+        async def startup_event():
+            fastapi_app.SEND_TASK = asyncio.create_task(self.consumer())
+            self._server_ready = True
+
+        @fastapi_app.on_event("shutdown")
+        def shutdown_event():
+            fastapi_app.SEND_TASK.cancel()
+            self._server_ready = False
+
+        def authenticate_private_endpoint(credentials: HTTPBasicCredentials = Depends(security)):
+            AUTO_SCALER_AUTH_PASSWORD = os.environ.get("AUTO_SCALER_AUTH_PASSWORD", "")
+            if len(AUTO_SCALER_AUTH_PASSWORD) == 0:
+                logger.warn("You have not set password for private endpoints!")
```

Comment:
can we please change this to: 

```suggestion
                logger.warn("You have not set password for private endpoints! Add --env AUTO_SCALER_AUTH_PASSWORD=<your pass> to your lightning run command.")
```